src/compas/geometry/shapes/polyhedron.py

```diff
@@ -263,16 +267,49 @@ def from_planes(cls, planes):
     # methods
     # ==========================================================================
 
-    def to_vertices_and_faces(self):
+    def to_vertices_and_faces(self, triangulated=False):
         """Returns a list of vertices and faces.
 
+        Parameters
+        ----------
+        triangulated: bool, optional
+            Flag indicating that the faces have to be triangulated.
+
         Returns
         -------
         (vertices, faces)
             A list of vertex locations and a list of faces,
             with each face defined as a list of indices into the list of vertices.
         """
-        return self.vertices, self.faces
+        if triangulated:
+            vertices = self.vertices[:]
+            faces = []
+            for face in self.faces:
+                if len(face) > 3:
+                    centroid = centroid_polygon([vertices[index] for index in face])
+                    index = len(vertices)
+                    vertices.append(centroid)
+                    for a, b in pairwise(face):
+                        faces.append([a, b, index])
+                else:
+                    faces.append(face)
+        else:
+            vertices = self.vertices
+            faces = self.faces
+        return vertices, faces
+
+    def is_closed(self):
+        """Verify that the polyhedron forms a closed surface.
+
+        Returns
+        -------
+        bool
+            True if the polyhedron is closed.
+            False otherwise.
+        """
+        from compas.datastructures import Mesh
+        mesh = Mesh.from_vertices_and_faces(self.vertices, self.faces)
+        return mesh.is_closed()
```

Comment:
not very critical, but in order to avoid the circular dependency here, perhaps the `is_closed` should be entirely copied in this class (considering it's only a couple of lines) or moved as a shared utility function somewhere else.

Reply:
it is true that it is only a few lines of code, but those depend on a lot of the data structure infrastructure so it is not so simple to copy it out